src/java/org/apache/cassandra/db/commitlog/CommitLogReader.java

```diff
@@ -476,6 +483,9 @@ protected void readMutation(CommitLogReadHandler handler,
 
         if (shouldReplay)
             handler.handleMutation(mutation, size, entryLocation, desc);
+
+        // if the commit log segment was previously added to this list due to invalid mutations, it will be removed for those table ids that were successful replayed
```

Comment:
A new `CommitLogReplayer` is created for every `recoverFiles()` invocation [here](https://github.com/datastax/cassandra/pull/1031/files#diff-d6f1176d5c065977873ef3adab879ef8e4e2a36ad5d0a2480f94b4c9f63d0cc3R266), so do we actually need this? Also why do we need to track the table ids at all?

Reply:
Yeah you're right - we do not need this one or to track table ids at all.